src/common/guac_rect.h

```diff
@@ -81,5 +103,40 @@ void guac_common_rect_extend(guac_common_rect* rect, const guac_common_rect* min
  */
 void guac_common_rect_constrain(guac_common_rect* rect, const guac_common_rect* max);
 
+/**
+ * Check whether a rectangle intersects another
+ *
+ * @param rect
+ *     Rectangle to check for intersection
+ *
+ * @param master
+ *     The master rectangle
```

Comment:
I don't immediately have a suggestion for improving this, but "the master rectangle" is not very clear either. There must be a more descriptive/clear way to document this. :confused: